models/glif_psc.cpp

```diff
@@ -0,0 +1,578 @@
+  updateValue< double >( d, names::b_reset, voltage_reset_b_ );
+
+  updateValue< double >( d, names::a_voltage, a_voltage_ );
+  updateValue< double >( d, names::b_voltage, b_voltage_ );
+
+  updateValue< std::vector< double > >( d, names::asc_init, asc_init_ );
+  updateValue< std::vector< double > >( d, names::k, k_ );
+  updateValue< std::vector< double > >( d, names::asc_amps, asc_amps_ );
+  updateValue< std::vector< double > >( d, names::r, r_ );
+
+  updateValue< std::vector< double > >( d, names::tau_syn, tau_syn_ );
+  updateValue< model_type >( d, names::glif_model, glif_model_ );
+
+  // check after ASC parameters size
+  if ( not(
+         ( asc_init_.size() == k_.size() ) && ( k_.size() == asc_amps_.size() ) && ( asc_amps_.size() == r_.size() ) ) )
+  {
+    throw BadProperty(
+      "All after spike current parameters (i.e., asc_init, k, asc_amps, r) "
+      "must have the same size." );
+  }
+
+  if ( V_reset_ >= th_inf_ )
+  {
+    throw BadProperty( "Reset potential must be smaller than threshold." );
+  }
+
+  if ( C_m_ <= 0.0 )
+  {
+    throw BadProperty( "Capacitance must be strictly positive." );
+  }
+
+  if ( G_ <= 0.0 )
+  {
+    throw BadProperty( "Membrane conductance must be strictly positive." );
+  }
+
+  if ( t_ref_ <= 0.0 )
+  {
+    throw BadProperty( "Refractory time constant must be strictly positive." );
+  }
+
+  if ( b_voltage_ <= 0.0 )
+  {
+    throw BadProperty( "Voltage-induced threshold time constant must be strictly positive." );
+  }
+
+  if ( b_spike_ <= 0.0 )
+  {
+    throw BadProperty( "Spike induced threshold time constant must be strictly positive." );
+  }
+
+  for ( std::size_t a = 0; a < k_.size(); ++a )
+  {
+    if ( k_[ a ] <= 0.0 )
+    {
+      throw BadProperty( "After-spike current time constant must be strictly positive." );
+    }
+  }
+
+  const size_t old_n_receptors = this->n_receptors_();
+  if ( updateValue< std::vector< double > >( d, names::tau_syn, tau_syn_ ) )
+  {
+    if ( this->n_receptors_() != old_n_receptors && has_connections_ == true )
+    {
+      throw BadProperty(
+        "The neuron has connections, therefore the number of ports cannot be "
+        "reduced." );
+    }
+    for ( size_t i = 0; i < tau_syn_.size(); ++i )
+    {
+      if ( tau_syn_[ i ] <= 0 )
+      {
+        throw BadProperty( "All synaptic time constants must be strictly positive." );
+      }
+    }
+  }
+  return delta_EL;
+}
+
+void
+nest::glif_psc::State_::get( DictionaryDatum& d, const Parameters_& p ) const
+{
+  def< double >( d, names::V_m, U_ + p.E_L_ );
+  def< std::vector< double > >( d, names::ASCurrents, ASCurrents_ );
+}
+
+void
+nest::glif_psc::State_::set( const DictionaryDatum& d, const Parameters_& p, double delta_EL )
+{
+  if ( updateValue< double >( d, names::V_m, U_ ) )
+  {
+    U_ -= p.E_L_;
+  }
+  else
+  {
+    U_ -= delta_EL;
+  }
+
+  updateValue< std::vector< double > >( d, names::ASCurrents, ASCurrents_ );
```

Comment:
Since you allow setting the ASCurrents, would it also make sense to allow setting Theta_s and Theta_v?
Should there be a protection against setting ASC or Theta for model variants not supporting them?

Reply:
ASC checking has been added.

Reply:
thta_s and theta_v are settable in the codes now.